server/src/main/java/org/elasticsearch/search/rescore/RescorePhase.java

```diff
@@ -39,15 +43,26 @@ public static void execute(SearchContext context) {
         if (context.size() == 0 || context.rescore() == null || context.rescore().isEmpty()) {
             return;
         }
-
+        assert validateSort(context.sort()) : "invalid sort";
```

Comment:
Maybe we could change this to a hard failure instead of an assertion, as it would make the following execution unpredictable ?